src/classes/BusinessLogicServices/ShopOrderService.php

```diff
@@ -0,0 +1,335 @@
+<?php
+
+namespace Packlink\PrestaShop\Classes\BusinessLogicServices;
+
+use Address as PrestaShopAddress;
+use Logeecom\Infrastructure\Logger\Logger;
+use Logeecom\Infrastructure\ORM\QueryFilter\QueryFilter;
+use Logeecom\Infrastructure\ORM\RepositoryRegistry;
+use Logeecom\Infrastructure\ServiceRegister;
+use Order as PrestaShopOrder;
+use Packlink\BusinessLogic\Configuration;
+use Packlink\BusinessLogic\Http\DTO\Shipment;
+use Packlink\BusinessLogic\Http\DTO\Tracking;
+use Packlink\BusinessLogic\Order\Exceptions\OrderNotFound;
+use Packlink\BusinessLogic\Order\Objects\Address;
+use Packlink\BusinessLogic\Order\Objects\Item;
+use Packlink\BusinessLogic\Order\Objects\Order;
+use Packlink\BusinessLogic\ShippingMethod\Interfaces\ShopShippingMethodService;
+use Packlink\PrestaShop\Classes\Entities\CartCarrierDropOffMapping;
+use Packlink\PrestaShop\Classes\Repositories\OrderRepository;
+use Packlink\PrestaShop\Classes\Utility\TranslationUtility;
+
+/**
+ * Class ShopOrderService
+ *
+ * @package Packlink\PrestaShop\Classes\BusinessLogicServices
+ */
+class ShopOrderService implements \Packlink\BusinessLogic\Order\Interfaces\ShopOrderService
+{
+    /**
+     * Handles updated tracking info for order with a given ID.
+     *
+     * @param string $orderId Shop order ID.
+     * @param Shipment $shipment Shipment object containing tracking codes and tracking url.
+     * @param Tracking[] $trackingHistory Shipment tracking history.
+     *
+     * @throws \Packlink\BusinessLogic\Order\Exceptions\OrderNotFound
+     */
+    public function updateTrackingInfo($orderId, Shipment $shipment, array $trackingHistory)
+    {
+        if (!empty($shipment->trackingCodes)) {
+            $repository = new OrderRepository();
+            $repository->setTrackingNumber((int)$orderId, $shipment->trackingCodes[0]);
+        }
+    }
+
+    /**
+     * Sets order Packlink shipping status to an order with a given ID.
+     *
+     * @param string $orderId Shop order ID.
+     * @param string $shippingStatus Packlink shipping status.
+     *
+     * @throws \Packlink\BusinessLogic\Order\Exceptions\OrderNotFound When order for provided reference is not found.
+     */
+    public function updateShipmentStatus($orderId, $shippingStatus)
+    {
+        $repository = new OrderRepository();
+
+        /** @var ConfigurationService $configService */
+        $configService = ServiceRegister::getService(Configuration::CLASS_NAME);
+        $statusMappings = $configService->getOrderStatusMappings();
+
+        if (!array_key_exists($shippingStatus, $statusMappings)) {
+            Logger::logWarning(
+                TranslationUtility::__('Order status mapping not found.'),
+                'Integration'
+            );
+
+            return;
```

Comment:
if you use return to break the flow , Why don't you use if-else?

Reply:
Exactly to avoid if-else when not needed.

Reply:
I think the code is not avoid if-else in this case code is implementing shadow if-else . If you want to avoid if-else will be better use other designs patterns . I think will be better to read code if-else instead a return into if clouse

Reply:
updated